src/dataSource.js

```diff
@@ -90,84 +87,201 @@ class DataSource {
+    let newDataRow = null;
+    let modifyRowData = null;
+
+    if (this.hot.hasHook('modifyRowData')) {
+      modifyRowData = this.hot.runHooks('modifyRowData', row);
+    }
+
+    dataRow = modifyRowData !== null && isNaN(modifyRowData) ? modifyRowData : this.data[row];
+
+    if (Array.isArray(dataRow)) {
+      newDataRow = [];
+
+      if (getAllProps) {
+        dataRow.forEach((cell, column) => {
+          newDataRow[column] = this.getAtPhysicalCell(row, column, dataRow);
+        });
+
+      } else {
+        // Only the columns from the provided range
+        rangeEach(startColumn, endColumn, (column) => {
+          newDataRow[column - startColumn] = this.getAtPhysicalCell(row, column, dataRow);
+        });
+      }
+
+    } else if (isObject(dataRow) || isFunction(dataRow)) {
+      if (toArray) {
+        newDataRow = [];
+      } else {
+        newDataRow = {};
+      }
+
+      if (!getAllProps || toArray) {
+        const rangeStart = 0;
+        const rangeEnd = this.countFirstRowKeys() - 1;
+
+        rangeEach(rangeStart, rangeEnd, (column) => {
+          const prop = this.colToProp(column);
+
+          if (column >= (startColumn || rangeStart) && column <= (endColumn || rangeEnd) && isNaN(prop)) {
+            const cellValue = this.getAtPhysicalCell(row, prop, dataRow);
+
+            if (toArray) {
+              newDataRow.push(cellValue);
+
+            } else {
+              setProperty(newDataRow, prop, cellValue);
+            }
+          }
+        });
+
+      } else {
+        objectEach(dataRow, (value, prop) => {
+          setProperty(newDataRow, prop, this.getAtPhysicalCell(row, prop, dataRow));
+        });
+      }
+    }
+    return newDataRow;
   }
 
   /**
-   * Returns a single value from the data.
+   * Set the provided value in the source data set at the provided coordinates.
    *
    * @param {number} row Physical row index.
-   * @param {number} column Visual column index.
-   * @returns {*}
+   * @param {number|string} column Property name / physical column index.
+   * @param {*} value The value to be set at the provided coordinates.
    */
-  getAtCell(row, column) {
-    let result = null;
+  setAtCell(row, column, value) {
+    if (row >= this.countRows() || column >= this.countFirstRowKeys()) {
+      // Not enough rows and/or columns.
+      return;
+    }
 
-    const modifyRowData = this.hot.runHooks('modifyRowData', row);
+    if (this.hot.hasHook('modifySourceData')) {
+      const valueHolder = createObjectPropListener(value);
 
-    const dataRow = isNaN(modifyRowData) ? modifyRowData : this.data[row];
+      this.hot.runHooks('modifySourceData', row, this.propToCol(column), valueHolder, 'set');
 
-    if (dataRow) {
-      const prop = this.colToProp(column);
+      if (valueHolder.isTouched()) {
+        value = valueHolder.value;
+      }
+    }
 
-      if (typeof prop === 'string') {
-        result = getProperty(dataRow, prop);
+    if (isNaN(column)) {
```

Comment:
Maybe check it by `Number.isInteger`? The `isNaN` is used also for checking `undefined`. For me, it would be more clear.